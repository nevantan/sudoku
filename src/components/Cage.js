import React from 'react';
import Cell from './Cell';

class Cage extends React.Component {
  constructor(props) {
    super(props);
  }
  componentWillMount = () => {
    const cage = this.props.cage;
    this.row = Math.floor(cage / 3) * 3;
    this.col = (cage - this.row) * 3;
  };
  render() {
    return (
      <div>
        <p>Cage Component</p>
        <Cell row={this.row} column={this.col} />
        <Cell row={this.row} column={this.col + 1} />
        <Cell row={this.row} column={this.col + 2} />
        <Cell row={this.row + 1} column={this.col} />
        <Cell row={this.row + 1} column={this.col + 1} />
        <Cell row={this.row + 1} column={this.col + 2} />
        <Cell row={this.row + 2} column={this.col} />
        <Cell row={this.row + 2} column={this.col + 1} />
        <Cell row={this.row + 2} column={this.col + 2} />
      </div>
    );
  }
}

export default Cage;
