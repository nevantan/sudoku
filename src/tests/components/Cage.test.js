import React from 'react';
import { shallow } from 'enzyme';
import Cage from '../../components/Cage';

test('should render Cage component correctly', () => {
  const wrapper = shallow(<Cage />);
  expect(wrapper).toMatchSnapshot();
});
