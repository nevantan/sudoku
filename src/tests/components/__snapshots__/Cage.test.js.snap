// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Cage component correctly 1`] = `
<div>
  <p>
    Cage Component
  </p>
  <Connect(Cell)
    column={NaN}
    row={NaN}
  />
  <Connect(Cell)
    column={NaN}
    row={NaN}
  />
  <Connect(Cell)
    column={NaN}
    row={NaN}
  />
  <Connect(Cell)
    column={NaN}
    row={NaN}
  />
  <Connect(Cell)
    column={NaN}
    row={NaN}
  />
  <Connect(Cell)
    column={NaN}
    row={NaN}
  />
  <Connect(Cell)
    column={NaN}
    row={NaN}
  />
  <Connect(Cell)
    column={NaN}
    row={NaN}
  />
  <Connect(Cell)
    column={NaN}
    row={NaN}
  />
</div>
`;
